From 860744b94a10a159562fc491fd7f3ea1388965c1 Mon Sep 17 00:00:00 2001
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Sat, 20 Feb 2021 11:00:55 +0800
Subject: [PATCH] firmware: arm_sdei: add interrupt binding api
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 860744b94a10a159562fc491fd7f3ea1388965c1


hulk inclusion
category: feature
bugzilla: 48046
CVE: NA

-------------------------------------------------------------------------

This patch add a interrupt binding api function which returns the binded
event number.

Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/firmware/arm_sdei.c | 10 ++++++++++
 include/linux/arm_sdei.h    |  1 +
 2 files changed, 11 insertions(+)

diff --git a/drivers/firmware/arm_sdei.c b/drivers/firmware/arm_sdei.c
index 840754dcc6ca..0dba8923cf20 100644
--- a/drivers/firmware/arm_sdei.c
+++ b/drivers/firmware/arm_sdei.c
@@ -187,6 +187,16 @@ int sdei_api_event_context(u32 query, u64 *result)
 }
 NOKPROBE_SYMBOL(sdei_api_event_context);
 
+int sdei_api_event_interrupt_bind(int hwirq)
+{
+	u64 event_number;
+
+	invoke_sdei_fn(SDEI_1_0_FN_SDEI_INTERRUPT_BIND, hwirq, 0, 0, 0, 0,
+			&event_number);
+
+	return (int)event_number;
+}
+
 static int sdei_api_event_get_info(u32 event, u32 info, u64 *result)
 {
 	return invoke_sdei_fn(SDEI_1_0_FN_SDEI_EVENT_GET_INFO, event, info, 0,
diff --git a/include/linux/arm_sdei.h b/include/linux/arm_sdei.h
index 0a241c5c911d..f809d0f703cf 100644
--- a/include/linux/arm_sdei.h
+++ b/include/linux/arm_sdei.h
@@ -36,6 +36,7 @@ int sdei_event_unregister(u32 event_num);
 
 int sdei_event_enable(u32 event_num);
 int sdei_event_disable(u32 event_num);
+int sdei_api_event_interrupt_bind(int hwirq);
 
 /* GHES register/unregister helpers */
 int sdei_register_ghes(struct ghes *ghes, sdei_event_callback *normal_cb,
-- 
2.26.2

